mcs/class/corlib/System/Array.cs

```diff
@@ -199,8 +202,10 @@ internal void InternalArray__set_Item<T> (int index, T item)
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
 		internal extern void SetGenericValueImpl<T> (int pos, ref T value);
 
-		internal struct InternalEnumerator<T> : IEnumerator<T>
+		internal class InternalEnumerator<T> : IEnumerator<T>
 		{
+			public static readonly InternalEnumerator<T> Empty = new InternalEnumerator<T> (Array.Empty <T> ());
```

Comment:
Out InternalEnumerator is not stateless and it cannot be called from multiple instances. It's probably best to implement a new class EmptyInternalEnumerator instead

Reply:
While the logic in InternalEnumerator would have been a no-op for empty enumerators, I made another class to keep things simple.